src/ui/components/TestSuite/suspense/TestEventDetailsCache.ts

```diff
@@ -321,39 +312,57 @@ async function fetchAndCachePossibleCypressDomNode(
             return cachedPropObject;
           })
         );
-        return yieldedDomNodes[0] ?? null;
+        return yieldedDomNodes ?? null;
       } else {
         return cachedPropObject;
       }
     })
   );
 
+  const flattenedDomNodes = possibleDomNodes.flat().filter(Boolean);
+
   // Try to make sure this looks like a DOM node
-  const firstDomNode = possibleDomNodes.find(
+  const matchingDomNodes = flattenedDomNodes.filter(
     el => !!el && el.className !== "Object" && el.preview?.node
   );
 
-  await cacheDomNodeEntry(firstDomNode, replayClient, pauseId, timeStampedPoint);
+  await cacheDomNodeEntry(matchingDomNodes, replayClient, pauseId, timeStampedPoint);
 }
 
 async function cacheDomNodeEntry(
-  firstDomNode: ProtocolObject | null | undefined,
+  matchingDomNodes: ProtocolObject[],
   replayClient: ReplayClientInterface,
   pauseId: string,
   timeStampedPoint: TimeStampedPoint
 ) {
-  let nodeInfo: Element | null = null;
+  let elements: Element[] = [];
 
-  if (firstDomNode) {
-    // Kick off a fetch for the bounding rects now, so we have that cached when we try to highlight this node
+  if (matchingDomNodes.length > 0) {
     boundingRectsCache.prefetch(replayClient, pauseId);
-    nodeInfo = await elementCache.readAsync(replayClient, pauseId, firstDomNode.objectId);
+    elements = await Promise.all(
+      matchingDomNodes.map(async domNode => {
+        const cachedDomNode = await elementCache.readAsync(replayClient, pauseId, domNode.objectId);
+        return cachedDomNode;
+      })
+    );
+
+    const uniqueDomNodeIds = new Set<string>();
+
+    // Only want to show elements that are actually in the page,
+    // and remove any duplicates
+    elements = elements.filter(e => {
+      if (e.node.isConnected && !uniqueDomNodeIds.has(e.id)) {
+        uniqueDomNodeIds.add(e.id);
```

Comment:
I'm curious about the duplicate element check.

Reply:
Some of the steps, like `eq` have both `"Applied To"` and `"Yielded"`, but the same node appears in each, like: `{"Applied To": ["a", "b", "c", "d"], "Yielded": "a"}`. That led to duplicate component key warnings when the highlighting logic tried to highlight `["a", "b", "c", "d", "a"]`.

Reply:
Hmm. Thinking about that further... maybe it makes more sense to pick _one_ of those.  "Yielded" would presumably be first, because that's what the step settled on.

Reply:
I think that makes sense.

Reply:
Did your thinking here change any or still considering it?

Reply:
Yeah, the last push changed to just pick one of "Yielded" or "Applied To", preferring "Yielded" first.